#  Easy
Нужно написать функцию, которая на вход получает список из чисел, а на выходе дает список,
где каждое число возведено в квадрат. Решать при помощи цикла.

# 1.
list(range(10))

In [16]:
list_1 = [value ** 2 for value in range(10)]
print(list_1)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


#  2.
list(range(100))

In [18]:
list_2 = [value ** 2 for value in range(100)]
print(list_2)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81, 100, 121, 144, 169, 196, 225, 256, 289, 324, 361, 400, 441, 484, 529, 576, 625, 676, 729, 784, 841, 900, 961, 1024, 1089, 1156, 1225, 1296, 1369, 1444, 1521, 1600, 1681, 1764, 1849, 1936, 2025, 2116, 2209, 2304, 2401, 2500, 2601, 2704, 2809, 2916, 3025, 3136, 3249, 3364, 3481, 3600, 3721, 3844, 3969, 4096, 4225, 4356, 4489, 4624, 4761, 4900, 5041, 5184, 5329, 5476, 5625, 5776, 5929, 6084, 6241, 6400, 6561, 6724, 6889, 7056, 7225, 7396, 7569, 7744, 7921, 8100, 8281, 8464, 8649, 8836, 9025, 9216, 9409, 9604, 9801]


# 3.
list()

In [6]:
list_3 =[value ** 2 for value in range()]
print(list_3)

TypeError: range expected at least 1 argument, got 0

# Normal
Тоже самое, но на этот раз без цикла (смотрим раздел с функциями из ноутбука, который разбирали в классе). Сравнить скорость выполнения первого и второго варианта.

In [4]:
def square(x):
    return x ** 2

list_one = [i for i in range(10)]
list(map(square, list_one))

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]

In [5]:
%%timeit

list(map(square, list_one))

4.37 µs ± 182 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)


In [7]:
def square_element(x):
    return x ** 2

list_two = [i for i in range(100)]
list(map(square_element, list_two))

[0,
 1,
 4,
 9,
 16,
 25,
 36,
 49,
 64,
 81,
 100,
 121,
 144,
 169,
 196,
 225,
 256,
 289,
 324,
 361,
 400,
 441,
 484,
 529,
 576,
 625,
 676,
 729,
 784,
 841,
 900,
 961,
 1024,
 1089,
 1156,
 1225,
 1296,
 1369,
 1444,
 1521,
 1600,
 1681,
 1764,
 1849,
 1936,
 2025,
 2116,
 2209,
 2304,
 2401,
 2500,
 2601,
 2704,
 2809,
 2916,
 3025,
 3136,
 3249,
 3364,
 3481,
 3600,
 3721,
 3844,
 3969,
 4096,
 4225,
 4356,
 4489,
 4624,
 4761,
 4900,
 5041,
 5184,
 5329,
 5476,
 5625,
 5776,
 5929,
 6084,
 6241,
 6400,
 6561,
 6724,
 6889,
 7056,
 7225,
 7396,
 7569,
 7744,
 7921,
 8100,
 8281,
 8464,
 8649,
 8836,
 9025,
 9216,
 9409,
 9604,
 9801]

In [10]:
%%timeit

list(map(square_element, list_two))

42.5 µs ± 4.55 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)


Скорость выполнения для list(range(10)) - 4.37 µs < для list(range(100)) - 42.2 µs